tests/tests.py

```diff
@@ -0,0 +1,97 @@
+import json
+from unittest import expectedFailure
+
+from django.core.urlresolvers import reverse, resolve
+from django.test import SimpleTestCase, RequestFactory
+
+from edtf.views import edtf_form, boolean_result, result_json
+
+
+class TestURLs(SimpleTestCase):
+
+    def test_edtf_form_url(self):
+        view = resolve(reverse('edtf:edtf_form')).func
+        self.assertEqual(view, edtf_form)
+
+    def test_results_json_url(self):
+        view = resolve(reverse('edtf:results_json')).func
+        self.assertEqual(view, result_json)
+
+
+class TestEdtfFormView(SimpleTestCase):
+
+    def setUp(self):
+        self.factory = RequestFactory()
+
+    def test_returns_200(self):
+        request = self.factory.get('/')
+        response = edtf_form(request)
+        self.assertEqual(response.status_code, 200)
+
+    def test_correct_template_used(self):
+        url = reverse('edtf:edtf_form')
+        response = self.client.get(url)
+        self.assertTemplateUsed(response, 'edtf/edtf_form.html')
+
+
+class TestBooleanResultView(SimpleTestCase):
+
+    def setUp(self):
+        self.factory = RequestFactory()
+
+    def test_returns_200(self):
+        request = self.factory.get('/', {'date': '', 'level': ''})
+        response = boolean_result(request)
+        self.assertEqual(response.status_code, 200)
+
+    def test_returns_400_without_date(self):
+        request = self.factory.get('/', {'level': '1'})
+        response = boolean_result(request)
+        self.assertEqual(response.status_code, 400)
+
+    # The view should return an HTTP 400 response when no level is present.
+    @expectedFailure
+    def test_returns_400_without_level(self):
```

Comment:
The way I read boolean_result(), it should return a 200 if there is a date but no level. level is not required. This view is weird cause it isn't actually used and if theoretically you do submit with a level, this view doesn't make much use of it.